src/inject/dynamic-theme/css-rules.ts

```diff
@@ -58,22 +58,25 @@ export function iterateCSSDeclarations(style: CSSStyleDeclaration, iterate: (pro
         }
         iterate(property, value);
     });
-    if (isSafari && style.cssText.includes('var(')) {
-        // Safari doesn't show shorthand properties' values
-        shorthandVarDepPropRegexps.forEach(([prop, regexp]) => {
-            const match = style.cssText.match(regexp);
-            if (match && match[1]) {
-                const val = match[1].trim();
-                iterate(prop, val);
-            }
-        });
-    } else {
-        shorthandVarDependantProperties.forEach((prop) => {
-            const val = style.getPropertyValue(prop);
-            if (val && val.includes('var(')) {
-                iterate(prop, val);
-            }
-        });
+    if (isSafari) {
```

Comment:
If not Safari, then do nothing :) Nice optimisation :)

Seriously, it's hard to predict what exact style properties are more common. Here it seems like a bit more performant, but with other properties it will be a bit less performant. https://jsben.ch/CfD6l

Maybe some day we will grab some statistics, e.g. from top 100 sites. Anyway this exact check should not be a bottleneck, I assume it takes some 5%.

Worth comparing if it makes any noticeable difference in Gmail though.

Reply:
> Maybe some day we will grab some statistics, e.g. from top 100 sites. Anyway this exact check should not be a bottleneck, I assume it takes some 5%.

Having some kind of benchmark against real-word data is good in general. As we now have to rely on stripped down test cases which can or cannot have any meaning against real-world sites.

About Gmail, it still seems like a good improvement to have. Especially as once again, it's a popular function so even 1ms shaves of ~30ms in general. so if it shaves of ~5ms you will have shave of  ~150ms etc.

> If not Safari, then do nothing :) Nice optimisation :)

Was this whole hack just targeted to safari? As I thought I did the `else {` check for `if (isSafari) { } else { }`. But doesn't seem like I've correctly coded that. As if it's just to safari this shortHandDep, then I say this PR is good to go.